M2/Macaulay2/m2/monideal.m2

```diff
@@ -21,13 +21,13 @@ newMonomialIdeal = (R,rawI) -> new MonomialIdeal from {
 monomialIdealOfRow := (i,m) -> newMonomialIdeal(ring m,rawMonomialIdeal(raw m, i))
 
 codimopts := { Generic => false }
-codim MonomialIdeal := {  } >> opts -> m -> rawCodimension raw m
-codim Module := codimopts >> opts -> (cacheValue (symbol codim => opts)) (M -> runHooks(Module, symbol codim, (opts,M)))
+codim MonomialIdeal := codimopts >> opts -> m -> rawCodimension raw m
```

Comment:
This goes back to #1589. Here's the pitfall that I needed to avoid here:
```m2
i1 : genericFoo = method();

i2 : genericFoo Ideal := I -> 2 + codim(I, Generic => true);

i3 : R = kk[x];

i4 : genericFoo ideal x

o4 = 3

i5 : genericFoo monomialIdeal x
stdio:2:30:(3):[1]: error: encountered an unknown key or option: Generic
```
If MonomialIdeal inherits from Ideal, then it needs to inherit all of its methods as well.